tools/vscode/tasks.json

```diff
@@ -0,0 +1,334 @@
+            },
+            {
+              "regexp": "^(.*)$",
+              "message": 1
+            },
+            {
+              "regexp": "^\\s+at\\s.*\\s\\((\\/.*):(.*):(.*)\\)$",
+              "file": 1,
+              "line": 2,
+              "column": 3
+            }
+          ]
+        },
+        {
+          "owner": "sync",
+          "severity": "warning",
+          "fileLocation": [
+            "relative",
+            "${workspaceFolder}/.."
+          ],
+          "pattern": [
+            {
+              "regexp": "^WARNING:\\s'(.*)'\\s(.*)$",
+              "kind": "file",
+              "file": 1,
+              "message": 2
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "label": "generate_compile_commands",
+      "type": "shell",
+      "detail": "generate_compdb.py",
+      "command": "${workspaceFolder}/tools/clang/scripts/generate_compdb.py",
+      "args": [
+        "-p",
+        "${config:chrome.outputDir}",
+        "-o",
+        "compile_commands.json"
+      ],
+      "presentation": {
+        "reveal": "never",
+        "close": true
+      },
+      "options": {
+        "cwd": "${workspaceFolder}"
+      }
+    },
+    {
+      "label": "sync_and_generate",
+      "detail": "npm run sync && generate_compile_commands",
+      "dependsOn": [
+        "sync",
+        "generate_compile_commands"
+      ],
+      "presentation": {
+        "showReuseMessage": false
+      },
+      "dependsOrder": "sequence"
+    },
+    //   {
+    //     "label": "3-test_current_file",
+    //     "type": "shell",
+    //     "command": "${workspaceFolder}/tools/autotest.py -C ${config:chrome.outputDir} --run-all ${file}"
+    //   },
+    //   {
+    //     "label": "4-test_current_file_with_filter",
+    //     "type": "shell",
+    //     "command": "${workspaceFolder}/tools/autotest.py -C ${config:chrome.outputDir} --gtest_filter ${input:gtestFilter} ${file}"
+    //   },
+    //   {
+    //     "label": "5-test_current_line",
+    //     "type": "shell",
+    //     "command": "${workspaceFolder}/tools/autotest.py -C ${config:chrome.outputDir} --line ${lineNumber} ${file}"
+    //   },
+    //   {
+    //     "label": "6-test_current_directory",
+    //     "type": "shell",
+    //     "command": "${workspaceFolder}/tools/autotest.py -C ${config:chrome.outputDir} --run-all ${fileDirname}"
+    //   },
+    {
+      "label": "brave_unit_tests",
+      "type": "shell",
+      "group": "test",
+      "command": "npm run test -- brave_unit_tests -- -C ${config:chrome.outputDir}",
+      "presentation": {
+        "group": "test",
+        "reveal": "never"
+      },
+      "detail": "npm run test -- brave_unit_tests"
+    },
+    {
+      "label": "brave_unit_tests with filter",
+      "type": "shell",
+      "group": "test",
+      "command": "npm run test -- brave_unit_tests --filter=${input:gtestFilter} -C ${config:chrome.outputDir}",
```

Comment:
vscode input machinery is not very usable when you need to pass many test arguments such as `no-sandbox, test_launcher_jobs, enable-pixel-output-in-tests, ...` and vary them from time to time.

what about creating a file and passing lines from it into a test binary via `npm run test`? we can add a vscode command like "edit brave_unit_tests args" that will just open this file to edit (make it json5 with comments, etc.), it would be much better.

my args for `brave_browser_tests` currently look like this:
```
      "args": [
        "--test_launcher_jobs=10",
        "--gtest_filter=*BraveContentSettingsBrowserTest*",
        //"--disable-features=NavigationThreadingOptimizations",
        //"--gtest_repeat=20",
        "--enable-pixel-output-in-tests",
        "--no-sandbox",
        //"--enable-logging=stderr",
        "--lang=en",
        //"--wait-for-debugger",
      ],
```

Reply:
to me that sounds like something that should be part of brave-core scripts and not vscode specific, wdyt?

Reply:
let's make the template ready to be extended then. something like:
```
"command": "npm run test -- brave_unit_tests",
"args": [
  "-C ${config:chrome.outputDir}",
  "--filter=${input:gtestFilter}",
]
```

and do it also for browser tests.

Reply:
I think I'm actually going to remove this one because there's no way to save the last entry so you have to enter the filter every time :(

Reply:
instead of being useful, it's actually a hassle and easier to just run the last command in the terminal. Even re-run last task doesn't preserve it and I tried different options for reevaluate

Reply:
yep, this is another reason why I wanted to do it via an external file. vscode inputs for some reason are pretty unusable :(